android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java

```diff
@@ -150,7 +154,32 @@ private LinearLayout buildRepeatingGroupLayout(ViewParent parent) throws Excepti
                 }
                 // add element to json form object to be written into
                 JSONObject step = ((JsonApi) widgetArgs.getContext()).getmJSONObject().getJSONObject(widgetArgs.getStepName());
-                step.getJSONArray(FIELDS).put(element);
+                /*
+                Required to add Repeating group section just after repeating group
+                So that when you comeback to repeated group section step, all child sections
+                should be visible after that Repeating group. i.e. should add all child just after
+                repeating group field.
+                FIXED: Should add repeating group sections after repeating group field, not at end of
+                step.
+                 */
+                JSONArray fields = step.getJSONArray(FIELDS);
+                int repeatingGroupIndex = 0;
+                JSONArray updatedFields = new JSONArray();
+                String repeatingGroupKey = widgetArgs.getJsonObject().getString(KEY);
+                for(int fieldIndex = 0; fieldIndex < fields.length(); fieldIndex++) {
+                    JSONObject fieldObject = fields.getJSONObject(fieldIndex);
+                    updatedFields.put(fieldObject);
+                    if(fieldObject.has(KEY) && fieldObject.getString(KEY).equals(repeatingGroupKey)) {
```

Comment:
Prefer assigning `fieldObject.optString(KEY) ` to a variable and check if its `null`  and equal to `repeatingGroupKey`.